core/storage/learner_group/gae_models.py

```diff
@@ -0,0 +1,297 @@
+            elif user_id in learner_group_model.invited_user_ids:
+                user_data[learner_group_model.id] = {
+                    'title': learner_group_model.title,
+                    'description': learner_group_model.description,
+                    'facilitator_user_id': '',
+                    'student_user_id': '',
+                    'invited_user_id': user_id,
+                    'subtopic_page_ids':
+                        learner_group_model.subtopic_page_ids,
+                    'story_ids': learner_group_model.story_ids
+                }
+
+            # If the user is the facilitator of the group, we export all
+            # fields except student_user_ids, invited_user_ids and the
+            # facilitator_user_id field is exported only containing the
+            # current user's id.
+            elif user_id in learner_group_model.facilitator_user_ids:
+                user_data[learner_group_model.id] = {
+                    'title': learner_group_model.title,
+                    'description': learner_group_model.description,
+                    'facilitator_user_id': user_id,
+                    'student_user_id': '',
+                    'invited_user_id': '',
+                    'subtopic_page_ids':
+                        learner_group_model.subtopic_page_ids,
+                    'story_ids': learner_group_model.story_ids
+                }
+        return user_data
+
+    @classmethod
+    def has_reference_to_user_id(cls, user_id: str) -> bool:
+        """Check whether LearnerGroupModel contains data of a given user.
+
+        Args:
+            user_id: str. The ID of the user whose data should be checked.
+
+        Returns:
+            bool. Whether any models refer to the given user ID.
+        """
+        return (
+            cls.query(datastore_services.any_of(
+                cls.student_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitator_user_ids == user_id
+            )).get(keys_only=True) is not None
+        )
+
+    @classmethod
+    def apply_deletion_policy(cls, user_id: str) -> None:
+        """Delete all LearnerGroupModel instances associated with the
+        user.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to delete.
+        """
+        found_models = cls.get_all().filter(
+            datastore_services.any_of(
+                cls.student_user_ids == user_id,
+                cls.invited_user_ids == user_id,
+                cls.facilitator_user_ids == user_id
+        ))
+
+        for learner_group_model in found_models:
+            # If the user is a student, delete the user from the
+            # student_user_ids list.
+            if user_id in learner_group_model.student_user_ids:
+                learner_group_model.student_user_ids.remove(user_id)
+                learner_group_model.update_timestamps()
+                learner_group_model.put()
```

Comment:
Instead of putting one by one, rather modify the code to first modify all models and then put them all at once.

Reply:
Done!